### DISCUSSION TITLE
Mapping child type as parent, with explicit casting, maps derived properties

### DISCUSSION THREAD

**Discussion by Xriuk** [LuckyPennySoftware/AutoMapper#4286] [NA]
### Source/destination types

```csharp
public class Entity{
  public string MyId { get; set; }
  
  public string MyString { get; set; }
}


public class DtoEntityBase{
  public string MyString { get; set; }
}

public class DtoEntityDerived : DtoEntityBase {
  public string MyId { get; set; }
}
```

### Mapping configuration

```csharp
c.CreateMap<DtoEntityBase, Entity>()
  .ForMember(e => e.MyId, c => c.Ignore());
c.CreateMap<DtoEntityDerived, Entity>();

DtoEntityDerived a = new DtoEntityDerived{
  MyId = "AAA",
  MyString = "BBB"
};
Entity e = mapper.Map<DtoEntityBase, Entity>(a);
Console.WriteLine(e.MyId + " " + e.MyString);
```

### Expected behavior

Should print " BBB" as the MyId property should not be mapped

### Actual behavior

Prints "AAA BBB" as MyId property is being mapped despite the map types being DtoEntityBase to Entity

### Steps to reproduce
[Fiddle](https://dotnetfiddle.net/4LMsJC)

It seems that the passed types are ignored and actual types of the passed variables are used, is this by design?

**Comment by lbargaoanu** [comment]
I think this is better suited for StackOverflow.
